assets/js/theme/common/carousel/index.js

```diff
@@ -28,53 +30,45 @@ const extractSlidesDetails = ({
+    const { activeSlideIdx, slidesQuantity } = extractSlidesDetails($activeSlider[0].slick);
     const $carouselContentElement = $('.js-carousel-content-change-message', $parentContainer);
-    const carouselContentInitText = $carouselContentElement.text();
-    const carouselContentAnnounceMessage = updateTextWithLiveData(carouselContentInitText, (actualSlide + 1), actualSlideCount);
+    const carouselContentAnnounceMessage = updateTextWithLiveData(context.carouselContentAnnounceMessage, (activeSlideIdx + 1), slidesQuantity);
 
     $carouselContentElement.text(carouselContentAnnounceMessage);
 };
 
-export const onCarouselChange = (event, carousel) => {
+export const onCarouselChange = (e, carousel, context) => {
     const {
-        options: { prevArrow, nextArrow },
-        $prevArrow,
-        $nextArrow,
         $dots,
         $slider,
+        $prevArrow,
+        $nextArrow,
+        options: { infinite },
     } = carousel;
 
-    const { actualSlideCount, actualSlide } = extractSlidesDetails(carousel);
-    const $prevArrowNode = $prevArrow || $slider.find(prevArrow);
-    const $nextArrowNode = $nextArrow || $slider.find(nextArrow);
-
-    const dataArrowLabel = $slider.data('arrow-label');
+    const { activeSlideIdx, slidesQuantity } = extractSlidesDetails(carousel);
 
-    if (dataArrowLabel) {
-        $prevArrowNode.attr('aria-label', dataArrowLabel);
-        $nextArrowNode.attr('aria-label', dataArrowLabel);
-        $slider.data('arrow-label', false);
-    }
-
-    dotsSetup($dots, actualSlide, actualSlideCount, $slider.data('dots-labels'));
-    setTabindexes($slider.find('.slick-slide'), $prevArrowNode, $nextArrowNode, actualSlide, actualSlideCount);
-    arrowAriaLabling($prevArrowNode, $nextArrowNode, actualSlide, actualSlideCount);
-    tooltipSetup($prevArrowNode, $nextArrowNode, $dots);
+    dotsSetup($dots, activeSlideIdx, slidesQuantity, context);
+    setTabindexes($slider.find('.slick-slide'), $prevArrow, $nextArrow, activeSlideIdx, slidesQuantity, infinite);
+    arrowAriaLabling($prevArrow, $nextArrow, activeSlideIdx, slidesQuantity, context);
+    tooltipSetup($prevArrow, $nextArrow, $dots);
+    activatePlayPauseButton(carousel, slidesQuantity, context);
 };
 
-export default function () {
-    const $carouselCollection = $('[data-slick]');
-
-    if ($carouselCollection.length === 0) return;
-
-    $carouselCollection.each((index, carousel) => {
+export default function (context) {
+    $('[data-slick]').each((idx, carousel) => {
         // getting element using find to pass jest test
         const $carousel = $(document).find(carousel);
-        $carousel.on('init afterChange', onCarouselChange);
-        $carousel.on('click', '.slick-arrow, .js-carousel-dot', $carousel, onCarouselClick);
+        $carousel.on('init afterChange', (e, carouselObj) => onCarouselChange(e, carouselObj, context));
+        $carousel.on('click', '.slick-arrow, .slick-dots', $carousel, e => onCarouselClick(e, context));
```

Comment:
Could we pass direct message to onCarouselClick instead of context obj?

Reply:
As we discussed when util needs only one context value destructuring was replaced with relevant data